pack/nodejs/helm.ts

```diff
@@ -0,0 +1,715 @@
+    // expanded (default ".").
+    untardir?: string;
+
+    // Chart repository username.
+    username?: string;
+
+    // Location of your Helm config. Overrides $HELM_HOME (default "/Users/alex/.helm").
+    home?: string;
+
+    // Use development versions, too. Equivalent to version '>0.0.0-0'. If --version is set, this is
+    // ignored.
+    devel?: boolean;
+
+    // Fetch the provenance file, but don't perform verification.
+    prov?: boolean;
+
+    // If set to false, will leave the chart as a tarball after downloading.
+    untar?: boolean;
+
+    // Verify the package against its signature.
+    verify?: boolean;
+}
+
+// Retrieve a package from a package repository, and download it locally.
+//
+// This is useful for fetching packages to inspect, modify, or repackage. It can also be used to
+// perform cryptographic verification of a chart without installing the chart.
+//
+// There are options for unpacking the chart after download. This will create a directory for the
+// chart and uncompress into that directory.
+//
+// If the `verify` option is specified, the requested chart MUST have a provenance file, and MUST
+// pass the verification process. Failure in any part of this will result in an error, and the chart
+// will not be saved locally.
+export function fetch(chart: string, opts?: FetchOpts) {
+    const flags = opts != null ? [
+        // Untar by default.
+        ...opts.untar == false ? [] : [`--untar`],
+
+        ...opts.version != null     ? [`--version ${shell.quote([opts.version])}`]         : [],
+        ...opts.caFile != null      ? [`--ca-file ${shell.quote([opts.caFile])}`]          : [],
+        ...opts.certFile != null    ? [`--cert-file ${shell.quote([opts.certFile])}`]      : [],
+        ...opts.keyFile != null     ? [`--key-file ${shell.quote([opts.keyFile])}`]        : [],
+        ...opts.destination != null ? [`--destination ${shell.quote([opts.destination])}`] : [],
+        ...opts.keyring != null     ? [`--keyring ${shell.quote([opts.keyring])}`]         : [],
+        ...opts.password != null    ? [`--password ${shell.quote([opts.password])}`]       : [],
+        ...opts.repo != null        ? [`--repo ${shell.quote([opts.repo])}`]               : [],
+        ...opts.untardir != null    ? [`--untardir ${shell.quote([opts.untardir])}`]       : [],
+        ...opts.username != null    ? [`--username ${shell.quote([opts.username])}`]       : [],
+        ...opts.home != null        ? [`--home ${shell.quote([opts.home])}`]               : [],
+        ...opts.devel == true       ? [`--devel`]                                          : [],
+        ...opts.prov == true        ? [`--prov`]                                           : [],
+        ...opts.verify == true      ? [`--verify`]                                         : [],
+    ] : [];
+    execSync(`helm fetch ${chart} ${flags.join(" ")}`);
+}
+
+export namespace transform {
```

Comment:
I think we can get rid of this namespace.  It shouldn't be part of a Helm package here.  And I think generally these helpers (if even really needed) can be in another layer.

Reply:
I am pretty confident everyone who is doing any kind of transformation on any Chart at all will end up using one of these three utility functions. To me this is a compelling reason to keep it in.

Reply:
Is the problem that it's not readable?

Reply:
I feel like making it clearer what the primitive is, even if it results in a little bit of boiler plate, is going to make this a lot more consumable than defining/using combinators here.  

In particular, the one used in the current sample above would be quite a bit easier to read if it was just a callback that had an `if` in it's body.